Tests/CoreTests/ObjectiveCIRTests.swift

```diff
@@ -155,4 +155,72 @@ class ObjectiveCIRTests: XCTestCase {
         ] }
         XCTAssertEqual(expected, actual)
     }
+
+    func testEnumIntegralTypes() {
```

Comment:
General testing nit, prefer smaller similar tests over one that tests everything.

I would create a test class that has smaller tests in it for these different Integral type cases. That way if one fails during a refactor / code change it's clear what exactly broke.